Intersect (Core)/GameObjects/NpcBase.cs

```diff
@@ -30,6 +29,24 @@ public class NpcBase : DatabaseObject<NpcBase>, IFolderable
 
         [NotMapped] public int[] VitalRegen = new int[(int) Vitals.VitalCount];
 
+        [NotMapped]
+        public Dictionary<StatusTypes, bool> Immunities = new Dictionary<StatusTypes, bool>();
```

Comment:
I didn't realize this before but this could totally just be a list of immunities. If it's not in the list, it's not immune. Checkboxes for immunities can just add/remove the immunity from the list.

Caveat: when serialized/checked for differences the immunities should be sorted (by name or enum-erical value, I don't care, just so long as they are consistent so that even if someone unchecked and rechecks all the same things in a different order it doesn't consider it a change).

Reply:
Now a list